plugins/modules/aws_s3.py

```diff
@@ -554,14 +534,15 @@ def download_s3file(module, s3, bucket, obj, dest, retries, version=None):
             key = s3.get_object(Bucket=bucket, Key=obj, VersionId=version)
         else:
             key = s3.get_object(Bucket=bucket, Key=obj)
-    except botocore.exceptions.ClientError as e:
-        if e.response['Error']['Code'] == 'InvalidArgument' and 'require AWS Signature Version 4' in to_text(e):
+    except is_boto3_error_code(['404', '403']) as e:
+        # AccessDenied errors may be triggered if 1) file does not exist or 2) file exists but
+        # user does not have the s3:GetObject permission. 404 errors are handled by download_file().
+        module.fail_json_aws(e, msg="Could not find the key %s." % obj)
+    except is_boto3_error_code('InvalidArgument') as e:
+        if 'require AWS Signature Version 4' in to_text(e):
```

Comment:
Could we handle these with `is_boto3_error_message` instead of `to_text(e)`?

Reply:
Now that it's merged, yes :)